tensorbay/opendataset/COCO2017/loader.py

```diff
@@ -0,0 +1,225 @@
+#!/usr/bin/env python3
+#
+# Copyright 2021 Graviti. Licensed under MIT License.
+#
+# pylint: disable=invalid-name, missing-module-docstring
+
+import json
+import os
+from collections import defaultdict
+from typing import Any, DefaultDict, Dict, List, Tuple
+
+import numpy as np
+
+from ...dataset import Data, Dataset
+from ...label import (
+    Label,
+    LabeledBox2D,
+    LabeledKeypoints2D,
+    LabeledMultiPolygon,
+    LabeledRLE,
+    PanopticMask,
+)
+from ...utility.itertools import chunked
+from .._utility.glob import glob
+
+try:
+    from PIL import Image
+except ModuleNotFoundError:
+    from .._utility.mocker import Image
+
+DATASET_NAME = "COCO2017"
+
+
+def COCO2017(path: str) -> Dataset:
+    """Dataloader of COCO2017 open dataset.
+
+    .. _COCO2017: https://cocodataset.org/#home
+
+    The file structure should be like::
+
+        <path>
+            annotations/
+                panoptic_train2017/
+                    000000116037.png
+                    000000116040.png
+                    ...
+
+                panoptic_val2017/
+                test_panoptic_val2017/
+                instances_train2017.json
+                instances_val2017.json
+                panoptic_train2017.json
+                panoptic_val2017.json
+                person_keypoints_train2017.json
+                person_keypoints_val2017.json
+
+            train2017/
+                000000116026.jpg
+                000000116031.jpg
+
+            test2017/
+            val2017/
+            unlabeled2017/
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Returns:
+        Loaded :class: `~tensorbay.dataset.dataset.Dataset` instance.
+
+    """
+    root_path = os.path.abspath(os.path.expanduser(path))
+    dataset = Dataset(DATASET_NAME)
+    dataset.load_catalog(os.path.join(os.path.dirname(__file__), "catalog.json"))
+
+    for segment_name in ("train", "val"):
```

Comment:
shall we set `("train", "val")` and `("test", "unlabeled")` as global variables?